common/djangoapps/enrollment/views.py

```diff
@@ -597,15 +597,22 @@ def post(self, request):
 
             enterprise_course_consent = request.data.get('enterprise_course_consent')
             explicit_linked_enterprise = request.data.get('linked_enterprise_customer')
-            if has_api_key_permissions and enterprise_enabled():
+            if (enterprise_course_consent or explicit_linked_enterprise) and has_api_key_permissions and enterprise_enabled():
```

Comment:
I think it would be good to move the `enterprise_enabled()` check to an if statement that wraps all of this enterprise-related code, starting on line 598. It would be even better to move this enterprise-related code to a function in the enterprise support module that can be called here.

Reply:
Argh, so sorry, I merged this and immediately after this comment came.

Reply:
Is it urgent enough to call for a revert?

Reply:
No worries. It was just a suggestion, not a requirement. :)